js/common.js

```diff
@@ -1135,3 +1135,41 @@ function getUuidV4() {
       return v.toString(16);
    });
 }
+
+/** Track input changes and warn the user of unsaved changes if they try to navigate away */
+window.glpiUnsavedFormChanges = false;
+$(document).ready(function() {
+   // Try to limit tracking to item forms by binding to inputs under glpi_tabs only.
+   // Forms must have the data-track-changes attribute set to true.
+   // Form fields may have their data-track-changes attribute set to false to override the tracking on that input.
+   var glpiTabs = $('#page .glpi_tabs');
+   glpiTabs.on('input', 'form[data-track-changes="true"] input:not([data-track-changes="false"]),' +
+      'form[data-track-changes="true"] textarea:not([data-track-changes="false"])', function() {
+      window.glpiUnsavedFormChanges = true;
+   });
+   glpiTabs.on('change', 'form[data-track-changes="true"] select:not([data-track-changes="false"])', function() {
+      window.glpiUnsavedFormChanges = true;
+   });
+   glpiTabs.on('select2:select', 'form[data-track-changes="true"] select:not([data-track-changes="false"])', function() {
+      window.glpiUnsavedFormChanges = true;
+   });
+   $(window).on('beforeunload', function() {
+      if (window.glpiUnsavedFormChanges) {
+         // Only used for older browsers. Newer ones will display a localized message that is unique to the browser.
+         return "Do you want to leave this site? Changes you made may not be saved.";
```

Comment:
```suggestion
         return __("Do you want to leave this site? Changes you made may not be saved.");
```

Reply:
I checked browser support for adding their own localized message instead of a defined custom one. Firefox and Chrome display their own localized message since 2016 and so does IE 11 and Edge. Is our own localized string needed since no user who is using one of the browsers supported by GLPI will ever see it? It could just be an empty string that is returned.

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event :
- you can return an empty string as modern browsers will not use returned value as a message,
- you should call `preventDefault()` on the event.

I tested it on Chrome and it works.